xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAEnumeratorHD.cpp

```diff
@@ -546,238 +546,199 @@ ComPtr<ID3D11VideoProcessorOutputView> CEnumeratorHD::CreateVideoProcessorOutput
+  if (args.m_colorPrimaries == AVCOL_PRI_BT2020 && args.m_colorTransfer == AVCOL_TRC_ARIB_STD_B67)
   {
-    CLog::LogF(LOGWARNING,
-               "ID3D11VideoProcessorEnumerator1 required to evaluate support is not available.");
-    return {};
+    if (args.m_hdrOutput)
+    {
+      CLog::LogF(LOGDEBUG, "mapping exception for HLG > HDR");
+      inputArgs.color_transfer = AVCOL_TRC_SMPTE2084;
+    }
+    else
+    {
+      CLog::LogF(LOGDEBUG, "mapping exception for HLG > SDR");
+      inputArgs.color_transfer = AVCOL_TRC_BT709;
+    }
   }
 
-  // There are no full range equivalents of DXGI_COLOR_SPACE_YCBCR_STUDIO_G2084_TOPLEFT_P2020 / LEFT
-  // => exit early as unsupported.
-  if (isSourceFullRange)
-    return {};
+  inputCS = AvToDxgiColorSpace(inputArgs);
+  outputCS = AvToDxgiColorSpace(outputArgs);
 
-  // Top left chroma siting is always preferred for HDR because is used in UHD Blu-Rays
-  // ffmpeg flags of the source can be missing of bad anyway
-  DXGI_COLOR_SPACE_TYPE inputCS = DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_TOPLEFT_P2020;
+  CLog::LogF(LOGDEBUG, "final mapping of source color space: {} dest color space: {}",
+             inputCS == DXGI_COLOR_SPACE_CUSTOM ? "NA" : DX::DXGIColorSpaceTypeToString(inputCS),
+             outputCS == DXGI_COLOR_SPACE_CUSTOM ? "NA" : DX::DXGIColorSpaceTypeToString(outputCS));
 
-  const DXGI_COLOR_SPACE_TYPE destCS = DX::Windowing()->UseLimitedColor()
-                                           ? DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P709
-                                           : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+  if (inputCS == DXGI_COLOR_SPACE_CUSTOM || outputCS == DXGI_COLOR_SPACE_CUSTOM)
+    return {};
 
-  const ProcessorConversions convTopLeft =
+  const ProcessorConversions conversions =
       ListConversions(m_input_dxgi_format, std::vector<DXGI_COLOR_SPACE_TYPE>{inputCS},
-                      RenderingOutputFormats, std::vector<DXGI_COLOR_SPACE_TYPE>{destCS});
-
-  CLog::LogF(LOGDEBUG, "BT.2020 input color spaces supported with SDR {} range output: {}: {}",
-             DX::Windowing()->UseLimitedColor() ? "limited" : "full",
-             DX::DXGIColorSpaceTypeToString(inputCS), !convTopLeft.empty() ? "yes" : "no");
+                      RenderingOutputFormats, std::vector<DXGI_COLOR_SPACE_TYPE>{outputCS});
 
-  if (!convTopLeft.empty())
+  if (conversions.empty())
   {
-    if (CServiceBroker::GetLogging().IsLogLevelLogged(LOGDEBUG) &&
-        CServiceBroker::GetLogging().CanLogComponent(LOGVIDEO))
-    {
-      for (const auto& c : convTopLeft)
-        CLog::LogF(LOGDEBUG, "supported conversion: {}", c.ToString());
-    }
-
-    CLog::LogF(LOGDEBUG, "Preferred UHD Blu-Ray top left chroma siting will be used.");
-    return convTopLeft;
+    CLog::LogF(LOGWARNING, "no conversions supported.");
   }
-
-  // Try left chroma siting as fall back - half a pixel is not much difference at 4k
-  inputCS = DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P2020;
```

Comment:
`DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P2020` this color space must be preserved because is used in all NVIDIA graphics in playback of regular 4K HDR10 sources to SDR display (tone mapped). Removing this causes render method DXVA is not usable for 4K HDR on these systems (all NVIDIA).

See: https://github.com/xbmc/xbmc/pull/23109#issuecomment-1500419723